src/proxy/shield.go

```diff
@@ -301,9 +302,9 @@ func GetClientID(r *http.Request, route utils.ProxyRouteConfig) string {
 	isConstIP := utils.IsConstellationIP(remoteAddr)
 	isConstTokenValid := constellation.CheckConstellationToken(r) == nil
 
-	if (UseForwardedFor && r.Header.Get("x-forwarded-for") != "") || 
+	if ((UseForwardedFor || utils.IsTrustedProxy(remoteAddr) ) && r.Header.Get("x-forwarded-for") != "") || 
 		 (isTunneledIp && isConstIP && isConstTokenValid) {
-		ip, _ := utils.SplitIP(r.Header.Get("x-forwarded-for"))
+		ip := strings.TrimSpace(strings.Split(r.Header.Get("X-Forwarded-For"), ",")[0])
```

Comment:
> strings.Split returns a []string , I am not sure why this is not throwing a compilation error to be honest

Because TrimSpace use only first item with  ```[0]``` ?


> Also do not remove utils.SplitIP it is required to remove the port from the IP

X-Forwarded-For doesn't contains port, just a list of IP (v4 or v6) separated by commas

Reply:
Source : https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For

<details><summary>Exemple</summary>
<p>

```
X-Forwarded-For: 2001:db8:85a3:8d3:1319:8a2e:370:7348
X-Forwarded-For: 203.0.113.195
X-Forwarded-For: 203.0.113.195, 2001:db8:85a3:8d3:1319:8a2e:370:7348
X-Forwarded-For: 203.0.113.195,2001:db8:85a3:8d3:1319:8a2e:370:7348,198.51.100.178
```

</p>
</details>